librarina: `librarian release` results in `panic: runtime error: invalid memory address or nil pointer dereference`

See output below

```
(py392) partheniou@partheniou-vm-3:~/git/google-cloud-python$ librarian release
2025/08/22 11:45:12 INFO librarian arguments=[release]
2025/08/22 11:45:12 INFO Temporary working directory dir=/tmp/librarian-20250822T114512Z
2025/08/22 11:45:12 INFO repo not specified, using current working directory as repo root path=/usr/local/google/home/partheniou/git/google-cloud-python
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x828c3c]

goroutine 1 [running]:
github.com/googleapis/librarian/internal/librarian.Run({0xa92240, 0xe1f240}, {0xc000022090, 0x1, 0x1})
        /usr/local/google/home/partheniou/git/librarian/internal/librarian/librarian.go:76 +0x23c
main.main()
        /usr/local/google/home/partheniou/git/librarian/cmd/librarian/main.go:31 +0x58
```

output of `git log`

```
partheniou@partheniou-vm-3:~/git/librarian$ git log
commit f0edfd6caf48a2bc09404ae447855265262c6f8e (HEAD -> main, origin/main, origin/HEAD)
```

--- a/internal/librarian/release_init.go
+++ b/internal/librarian/release_init.go
@@ -55,7 +55,6 @@ func init() {
 	fs := cmdInit.Flags
 	cfg := cmdInit.Config
 
-	addFlagAPISource(fs, cfg)
 	addFlagPush(fs, cfg)
 	addFlagImage(fs, cfg)
 	addFlagLibrary(fs, cfg)

--- a/internal/librarian/release_init.go
+++ b/internal/librarian/release_init.go
@@ -16,6 +16,7 @@ package librarian
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"log/slog"
 	"os"
@@ -221,6 +222,9 @@ func getChangeType(commit *conventionalcommits.ConventionalCommit) string {
 // cleanAndCopyGlobalAllowlist cleans the files listed in global allowlist in
 // repoDir, excluding read-only files and copies global files from outputDir.
 func cleanAndCopyGlobalAllowlist(cfg *config.LibrarianConfig, repoDir, outputDir string) error {
+	if cfg == nil {
+		return errors.New("librarian config is not setup")
+	}
 	for _, globalFile := range cfg.GlobalFilesAllowlist {
 		if globalFile.Permissions == config.PermissionReadOnly {
 			continue

--- a/internal/librarian/release_init.go
+++ b/internal/librarian/release_init.go
@@ -16,7 +16,6 @@ package librarian
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"log/slog"
 	"os"
@@ -223,7 +222,8 @@ func getChangeType(commit *conventionalcommits.ConventionalCommit) string {
 // repoDir, excluding read-only files and copies global files from outputDir.
 func cleanAndCopyGlobalAllowlist(cfg *config.LibrarianConfig, repoDir, outputDir string) error {
 	if cfg == nil {
-		return errors.New("librarian config is not setup")
+		slog.Info("librarian config is not setup, skip copying global allowlist")
+		return nil
 	}
 	for _, globalFile := range cfg.GlobalFilesAllowlist {
 		if globalFile.Permissions == config.PermissionReadOnly {

--- a/internal/librarian/release_init.go
+++ b/internal/librarian/release_init.go
@@ -55,6 +55,7 @@ func init() {
 	fs := cmdInit.Flags
 	cfg := cmdInit.Config
 
+	addFlagAPISource(fs, cfg)
 	addFlagPush(fs, cfg)
 	addFlagImage(fs, cfg)
 	addFlagLibrary(fs, cfg)
@@ -120,11 +121,12 @@ func (r *initRunner) runInitCommand(ctx context.Context, outputDir string) error
 	}
 
 	initRequest := &docker.ReleaseInitRequest{
-		Cfg:            r.cfg,
-		State:          r.state,
-		LibraryID:      r.cfg.Library,
-		LibraryVersion: r.cfg.LibraryVersion,
-		Output:         outputDir,
+		Cfg:             r.cfg,
+		State:           r.state,
+		LibrarianConfig: r.librarianConfig,
+		LibraryID:       r.cfg.Library,
+		LibraryVersion:  r.cfg.LibraryVersion,
+		Output:          outputDir,
 	}
 
 	if err := r.containerClient.ReleaseInit(ctx, initRequest); err != nil {

--- a/internal/librarian/release_init_test.go
+++ b/internal/librarian/release_init_test.go
@@ -124,6 +124,42 @@ func TestInitRun(t *testing.T) {
 			},
 		},
 		{
+			name: "run release init command without librarian config",
+			runner: &initRunner{
+				workRoot:        filepath.Join(t.TempDir(), "work-root"),
+				containerClient: &mockContainerClient{},
+				cfg: &config.Config{
+					Library: "example-id",
+				},
+				state: &config.LibrarianState{
+					Libraries: []*config.LibraryState{
+						{
+							ID: "another-example-id",
+							SourceRoots: []string{
+								"dir3",
+								"dir4",
+							},
+						},
+						{
+							ID: "example-id",
+							SourceRoots: []string{
+								"dir1",
+								"dir2",
+							},
+						},
+					},
+				},
+				repo: &MockRepository{
+					Dir: filepath.Join(t.TempDir(), "repo"),
+				},
+			},
+			files: map[string]string{
+				"file1.txt":      "",
+				"dir1/file2.txt": "",
+				"dir2/file3.txt": "",
+			},
+		},
+		{
 			name: "run release init command for all libraries",
 			runner: &initRunner{
 				workRoot:        filepath.Join(t.TempDir(), "work-root"),

--- a/internal/librarian/release_init_test.go
+++ b/internal/librarian/release_init_test.go
@@ -124,6 +124,44 @@ func TestInitRun(t *testing.T) {
 			},
 		},
 		{
+			name: "librarian config is not setup",
+			runner: &initRunner{
+				workRoot:        filepath.Join(t.TempDir(), "work-root"),
+				containerClient: &mockContainerClient{},
+				cfg: &config.Config{
+					Library: "example-id",
+				},
+				state: &config.LibrarianState{
+					Libraries: []*config.LibraryState{
+						{
+							ID: "another-example-id",
+							SourceRoots: []string{
+								"dir3",
+								"dir4",
+							},
+						},
+						{
+							ID: "example-id",
+							SourceRoots: []string{
+								"dir1",
+								"dir2",
+							},
+						},
+					},
+				},
+				repo: &MockRepository{
+					Dir: filepath.Join(t.TempDir(), "repo"),
+				},
+			},
+			files: map[string]string{
+				"file1.txt":      "",
+				"dir1/file2.txt": "",
+				"dir2/file3.txt": "",
+			},
+			wantErr:    true,
+			wantErrMsg: "librarian config is not setup",
+		},
+		{
 			name: "run release init command for all libraries",
 			runner: &initRunner{
 				workRoot:        filepath.Join(t.TempDir(), "work-root"),

--- a/internal/librarian/release_init_test.go
+++ b/internal/librarian/release_init_test.go
@@ -124,44 +124,6 @@ func TestInitRun(t *testing.T) {
 			},
 		},
 		{
-			name: "librarian config is not setup",
-			runner: &initRunner{
-				workRoot:        filepath.Join(t.TempDir(), "work-root"),
-				containerClient: &mockContainerClient{},
-				cfg: &config.Config{
-					Library: "example-id",
-				},
-				state: &config.LibrarianState{
-					Libraries: []*config.LibraryState{
-						{
-							ID: "another-example-id",
-							SourceRoots: []string{
-								"dir3",
-								"dir4",
-							},
-						},
-						{
-							ID: "example-id",
-							SourceRoots: []string{
-								"dir1",
-								"dir2",
-							},
-						},
-					},
-				},
-				repo: &MockRepository{
-					Dir: filepath.Join(t.TempDir(), "repo"),
-				},
-			},
-			files: map[string]string{
-				"file1.txt":      "",
-				"dir1/file2.txt": "",
-				"dir2/file3.txt": "",
-			},
-			wantErr:    true,
-			wantErrMsg: "librarian config is not setup",
-		},
-		{
 			name: "run release init command for all libraries",
 			runner: &initRunner{
 				workRoot:        filepath.Join(t.TempDir(), "work-root"),

